recipes/grpc/all/conanfile.py

```diff
@@ -93,7 +93,8 @@ def requirements(self):
         else:
             self.requires("abseil/20230125.3", transitive_headers=True, transitive_libs=True)
         self.requires("c-ares/1.19.1")
-        self.requires("openssl/[>=1.1 <4]")
+        # openssl latest avaliable lts is 3.0.x
+        self.requires("openssl/[>=1.1 <3.1]")
```

Comment:
how does this recipe already work with OpenSSL 3? PR with OpenSSL 3 was committed to master (so no release yet) just yesterday
https://github.com/grpc/grpc/pull/31256

Reply:
if the changes were merged it is because all the packages built for all ConanCenter supported configurations - so at the very least it is compatible at the source level and linking level.

From what I can see, other package managers have also convered to OpenSSL 3, even for grpc and certainly before this commit that landed yesterday. 

Chances are the APIs used are still in OpenSSL3 and that PR just moves to use newer ones in some instances? Also the PR does address some tests as well (which as we know, are not covered by Conan Center CI).